json_serializable/test/integration/integration_test.dart

```diff
@@ -304,6 +304,14 @@ void main() {
     expect(dayTypeEnumValues, ['no-good', 'rotten', 'very-bad']);
   });
 
+  test('enum serialization has correct type', () {
+    // ignore: omit_local_variable_types
+    final Issue1145Regression cls =
+        Issue1145Regression(status: {Issue1145RegressionEnum.gamma: true});
+    // Not Map<String?, dynamic>
+    expect(cls.toJson()['status'], const TypeMatcher<Map<String, dynamic>>());
```

Comment:
What is this before the change? I'm confused as to the actual new behavior...

Reply:
The new behaviour is to return Map<String, dynamic> correctly, hence the typematcher. The previous behaviour was returning Map<String?, dynamic> - which as you say is not permitted by the json spec.

Reply:
ah! cool!